### PR TITLE
ci: add CI workflow, test-writer agent, and pre-commit hooks

### PR DESCRIPTION
**Pull request by stivaugoin**
## Summary

- **CI workflow** (`.github/workflows/ci.yml`): Runs lint, format check, TypeScript build, and Vitest on PRs to main and pushes to main
- **Test-writer agent** (`.claude/agents/test-writer.md`): TDD-first agent that writes behavioral tests before implementation code exists
- **Testing standards update**: Added TDD section, replaced coverage percentage targets with regression-focused philosophy
- **Pre-commit hook**: Husky + lint-staged runs Prettier and ESLint on staged `.ts/.tsx` files before each commit
- **Cleanup**: Removed snapshot test from templates, fixed pre-existing formatting issue in `db/connection.ts`

## Test plan

- [ ] CI workflow triggers on this PR and all 4 steps pass (lint, format, build, test)
- [ ] Pre-commit hook runs on next commit in the repo
- [ ] All 520 tests pass (`pnpm vitest run`)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: .claude/skills/testing-standards/SKILL.md [hunk 1]
```diff
@@ -332,20 +352,22 @@ cargo test -- --nocapture           # show println! output
...
-| Auto-generated files | Excluded                               | —                                       |
+| Layer              | What to test                                              | Approach                                |
+| ------------------ | --------------------------------------------------------- | --------------------------------------- |
+| `src/db/**`        | All public CRUD functions — input/output behavior         | Integration tests with in-memory SQLite |
+| `src/managers/**`  | Orchestration logic and end-to-end workflows              | Integration tests with in-memory SQLite |
+| `src/components/**`| User interactions, error states, conditional rendering    | RTL behavior tests                      |
+| `src/hooks/**`     | Data flow and state transitions                           | Vitest + RTL                            |
+| `src/lib/**`       | Pure logic, edge cases, round-trip behavior               | Unit tests, no mocks                    |
+| Auto-generated     | Nothing — excluded                                        | —                                       |
```
[the remark this answers is not in the document]
Comment (stivaugoin) [reply .claude/skills/testing-standards/SKILL.md:366]: [115bd30](https://github.com/vata-apps/vata-app/pull/39/commits/115bd30985765e1f967a7856e9b3ceeb6db3828d)

File: .claude/skills/testing-standards/SKILL.md [hunk 2]
```diff
@@ -236,7 +256,7 @@ export function renderWithProviders(ui: React.ReactElement, options?: RenderOpti
 
 ---
 
-## 7. Rust / Tauri Tests
+## 8. Rust / Tauri Tests
```
[the remark this answers is not in the document]
Comment (stivaugoin) [reply .claude/skills/testing-standards/SKILL.md:259]: [ddd7478](https://github.com/vata-apps/vata-app/pull/39/commits/ddd74786d1ef303014c8279766d4ad4a3ad413dc)
